pkg/vm/opcodebench_test.go

```diff
@@ -74,714 +75,1578 @@ func opParamSlotsPushVM(op opcode.Opcode, param []byte, sslot int, slotloc int,
+				op:     op,
+				params: []byte{0},
+				items:  []any{stackitem.NewArray(getArrayWithDeep(currDeep))},
+			})
+			currDeep += stepDeep
+		}
+	}
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			benchOpcode(b, opParamPushVM(tc.op, tc.params, tc.items...))
+		})
+	}
 }
 
-func arrayOfOnes(size int) []stackitem.Item {
-	var elems = make([]stackitem.Item, size)
-	for i := range elems {
-		elems[i] = stackitem.Make(1)
+func BenchmarkPushInt(b *testing.B) {
+	type testCase struct {
+		name   string
+		op     opcode.Opcode
+		params []byte
+	}
+	var (
+		testCases    []testCase
+		opToNumBytes = map[opcode.Opcode]int{
+			opcode.PUSHINT8:   1,
+			opcode.PUSHINT16:  2,
+			opcode.PUSHINT32:  4,
+			opcode.PUSHINT64:  8,
+			opcode.PUSHINT128: 16,
+			opcode.PUSHINT256: 32,
+		}
+	)
+	for op, numBytes := range opToNumBytes {
+		buf := make([]byte, numBytes)
+		buf[numBytes-1] = 0x80
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %d", op, numBytes),
+			op:     op,
+			params: buf,
+		})
+	}
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			benchOpcode(b, opParamVM(tc.op, tc.params))
+		})
 	}
-	return elems
 }
 
-func arrayOfIfaces(size int) []any {
-	var elems = make([]any, size)
-	for i := range elems {
-		elems[i] = 1
+func BenchmarkPushData(b *testing.B) {
+	type testCase struct {
+		name   string
+		params []byte
+	}
+	var (
+		testCases []testCase
+		stepLen   = stackitem.MaxSize / countPoints
+		currLen   = stepLen
+	)
+	for range countPoints {
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %d", opcode.PUSHDATA4, currLen),
+			params: make([]byte, currLen),
+		})
+		currLen += stepLen
+	}
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			benchOpcode(b, opParamVM(opcode.PUSHDATA4, tc.params))
+		})
 	}
-	return elems
 }
 
-func bigMap() *stackitem.Map {
-	var m = stackitem.NewMap()
-	for i := range 1024 {
-		m.Add(stackitem.Make(i), stackitem.Make(i))
+func BenchmarkConvert(b *testing.B) {
+	type testCase struct {
+		name   string
+		params []byte
+		items  []any
+	}
+	var (
+		testCases        []testCase
+		stepArrayLen     = MaxStackSize / countPoints
+		stepByteArrayLen = stackitem.MaxSize / countPoints
+		currArrayLen     = stepArrayLen - 1
+		currByteArrayLen = stepByteArrayLen
+	)
+	for range countPoints {
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %s %s %d", opcode.CONVERT, stackitem.ArrayT, stackitem.StructT, currArrayLen),
+			params: []byte{byte(stackitem.StructT)},
+			items:  []any{make([]any, currArrayLen)},
+		})
+		currArrayLen += stepArrayLen
+	}
+	for range countPoints {
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %s %s %d", opcode.CONVERT, stackitem.ByteArrayT, stackitem.BufferT, currByteArrayLen),
+			params: []byte{byte(stackitem.BufferT)},
+			items:  []any{stackitem.NewByteArray(make([]byte, currByteArrayLen))},
```

Comment:
The price of `CONVERT(src, dst)` is dependent from the `src`/`dst` type. The difference is clear with wide Array/Struct items since `slices.Clone` is used. However, even the price of CONVERT applied to different primitive types is different: ByteArray/Buffer implies `bytes.Clone` usage whereas Integer or Bool related conversions are more light-weighted.

@roman-khimov, I think we need to check how significant this dependency is. May be later, after some initial proposal is composed.